script.js

```diff
@@ -1,4 +1,151 @@
+// there are 4 mode .
+// 1) player 1 roll . 2) player 1 pick to make the highest number.
+// 3) player 2 roll. 4) player 2 pick to make the highest number.
+var mode = 'player 1 roll dice';
+
+// store player 1 highest number
+var playerOneHighestNum = 0;
+
+var randomDice1 = 0;
+var randomDice2 = 0;
+
+// There are 2 dice.
+var diceRoll1 = function () {
+  var randomDiceNum = Math.floor(Math.random() * 6) + 1;
+  return randomDiceNum;
+};
+
+var diceRoll2 = function () {
+  var randomDiceNum2 = Math.floor(Math.random() * 6) + 1;
+  return randomDiceNum2;
+};
+
+// player one roll 2 dice logic
+var rollTwoDiceForPlayerOne = function () {
+  var message = '';
+  randomDice1 = 0;
+  randomDice2 = 0;
+  randomDice1 = diceRoll1();
+  randomDice2 = diceRoll2();
+  console.log(randomDice1);
+  console.log(randomDice2);
+  message = 'Welcome player 1.<br> You rolled '
+    + randomDice1 + ' for dice 1 and ' + randomDice2 + ' for dice 2 <br> choose the order of the dice';
+  return message;
+};
+
+// player one pick dice logic
+var playerOnePickDiceOneOrDiceTwo = function (input) {
+  var message = '';
+  playerOneHighestNum = 0;
+  // If player 1 pick dice 1
+  if (input == '1') {
+    playerOneHighestNum = randomDice1 + '' + randomDice2;
+    message = 'Player 1, you chose Dice 1 first.<br> Your number is ' + playerOneHighestNum + '. <br> It is now player 2 turn';
+    console.log('player 1');
+    console.log(playerOneHighestNum);
+    return message;
+  }
+  // if player 1 pick dice 2
+  if (input == '2') {
+    playerOneHighestNum = randomDice2 + '' + randomDice1;
+    message = 'Player 1, you chose Dice 2 first.<br> Your number is ' + playerOneHighestNum + '. <br> It is now player 2 turn';
+    console.log('player 1');
+    console.log(playerOneHighestNum);
+    return message;
+  }
+};
+
+// player two roll 2 dice logic
+var rollTwoDiceForPlayerTwo = function () {
+  var message = '';
+  randomDice1 = 0;
+  randomDice2 = 0;
+  randomDice1 = diceRoll1();
+  randomDice2 = diceRoll2();
+  console.log(randomDice1);
+  console.log(randomDice2);
+  message = 'Welcome player 2.<br> You rolled '
+     + randomDice1 + ' for dice 1 and ' + randomDice2 + ' for dice 2 <br> choose the order of the dice';
+  return message;
+};
+// player two pick dice logic
+var playerTwoPickDiceOneOrDiceTwo = function (input) {
```

Comment:
Why is this function separate from `playerOnePickDiceOneOrDiceTwo`? The logic is essentially the same, except the player is different! Please re-use!